bin/compile

```diff
@@ -140,10 +140,13 @@ create_build_env
 [ ! "$NPM_CONFIG_CACHE" ] && NPM_CONFIG_CACHE=$(mktemp -d -t npmcache.XXXXX)
 export YARN_CACHE_FOLDER NPM_CONFIG_CACHE
 
-if [[ "$YARN2" == "true" && "$NODE_MODULES_CACHE" == "true" ]]; then
-  warn "
-    WARNING- You are using Yarn2. The buildpack won't cache, because Yarn 2 does not provide node modules.
-    "
+if [[ "$YARN2" == "true" ]]; then
+  if [[ "$NODE_MODULES_CACHE" == "true" ]]; then
+    warn "
+      WARNING- You are using Yarn2. The buildpack won't cache, because Yarn 2 does not provide node modules.
+      "
+  fi
+  warn " You should be using a vendored version of Yarn. Heroku will ignore the engine from the package.json. "
```

Comment:
This is a good start! The only thing we need to do is detect if yarn is set in the engines - this is the line where we determine that: https://github.com/heroku/heroku-buildpack-nodejs/blob/master/bin/compile#L148. So we will probably want to have another `if` block in the that `install_bins` method to display this warning.